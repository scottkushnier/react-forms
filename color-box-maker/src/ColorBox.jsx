
function ColorBox({color, width="50px", height="50px", deleteFn}) {
    return (
        <div>
          <div className='box' style={{backgroundColor: color, width, height}}>
          </div>
          <button onClick={deleteFn}>X</button>
        </div>
    )
  }

  export default ColorBox;