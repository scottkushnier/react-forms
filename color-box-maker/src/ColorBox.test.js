
import { render } from "@testing-library/react";
import ColorBox from "./ColorBox";

test("colorbox smoke test", function () {
  render(<ColorBox  color='white' height='10px' width='10px'/>)
});

test("matches snapshot", function() {
    const {asFragment} = render(<ColorBox  color='blue' height='20px' width='20px'/>);
    expect(asFragment()).toMatchSnapshot();
  });

