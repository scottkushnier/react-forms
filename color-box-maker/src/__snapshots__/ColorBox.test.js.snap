// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches snapshot 1`] = `
<DocumentFragment>
  <div>
    <div
      class="box"
      style="background-color: blue; width: 20px; height: 20px;"
    />
    <button>
      X
    </button>
  </div>
</DocumentFragment>
`;
